File: netlify/functions/pause-all-ads.js
const fetch = require('node-fetch');

exports.handler = async (event, context) => {
  const headers = {
    'Content-Type': 'application/json',
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Access-Control-Allow-Methods': 'POST, OPTIONS'
  };

  if (event.httpMethod === 'OPTIONS') {
    return { statusCode: 200, headers, body: '' };
  }

  try {
    const metaToken = process.env.META_ACCESS_TOKEN;
    const adAccountId = process.env.META_AD_ACCOUNT_ID;

    if (!metaToken || !adAccountId) {
      return {
        statusCode: 403,
        headers,
        body: JSON.stringify({ error: 'Meta credentials not configured' })
      };
    }

    // Get campaigns
    const url = `https://graph.facebook.com/v21.0/${adAccountId}/campaigns`;
    const params = new URLSearchParams({
      access_token: metaToken,
      fields: 'id,status'
    });

    const response = await fetch(`${url}?${params}`);
    const campaigns = await response.json();

    let pausedCount = 0;

    for (const campaign of campaigns.data || []) {
      if (campaign.status === 'ACTIVE') {
        const pauseUrl = `https://graph.facebook.com/v21.0/${campaign.id}`;
        const pauseParams = new URLSearchParams({
          access_token: metaToken,
          status: 'PAUSED'
        });

        await fetch(pauseUrl, {
          method: 'POST',
          body: pauseParams
        });

        pausedCount++;
      }
    }

    return {
      statusCode: 200,
      headers,
      body: JSON.stringify({ paused: pausedCount })
    };

  } catch (error) {
    return {
      statusCode: 500,
      headers,
      body: JSON.stringify({ error: error.message })
    };
  }
};

File: netlify/functions/pause-all-ads.py
import json
import os
import requests

def handler(event, context):
    """Pause all active campaigns"""
    
    headers = {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Headers": "Content-Type",
        "Access-Control-Allow-Methods": "POST, OPTIONS"
    }
    
    if event.get("httpMethod") == "OPTIONS":
        return {"statusCode": 200, "headers": headers, "body": ""}
    
    try:
        meta_config = {
            "access_token": os.environ.get("META_ACCESS_TOKEN"),
            "ad_account_id": os.environ.get("META_AD_ACCOUNT_ID")
        }
        
        if not meta_config["access_token"]:
            return {
                "statusCode": 403,
                "headers": headers,
                "body": json.dumps({"error": "Meta credentials not configured"})
            }
        
        # Get campaigns
        url = f"https://graph.facebook.com/v21.0/{meta_config['ad_account_id']}/campaigns"
        response = requests.get(url, params={
            "access_token": meta_config["access_token"],
            "fields": "id,status"
        }, timeout=30)
        
        campaigns = response.json()
        paused_count = 0
        
        for campaign in campaigns.get("data", []):
            if campaign.get("status") == "ACTIVE":
                # Pause campaign
                pause_url = f"https://graph.facebook.com/v21.0/{campaign['id']}"
                requests.post(pause_url, data={
                    "access_token": meta_config["access_token"],
                    "status": "PAUSED"
                }, timeout=30)
                paused_count += 1
        
        return {
            "statusCode": 200,
            "headers": headers,
            "body": json.dumps({"paused": paused_count})
        }
        
    except Exception as e:
        return {
            "statusCode": 500,
            "headers": headers,
            "body": json.dumps({"error": str(e)})
        }
